src/asm/macro.c

```diff
@@ -145,13 +145,25 @@ uint32_t macro_GetUniqueID(void)
 
 char const *macro_GetUniqueIDStr(void)
 {
+	if (uniqueID == (uint32_t)-1) {
+		// We're at the top-level fstack, not inside a macro or loop;
+		// `\@` should not expand to anything
+		assert(!uniqueIDPtr);
+		return NULL;
+	} else if (uniqueID == 0) {
+		// We're in a context that hasn't expanded `\@` yet;
+		// assign it a new unique value
+		assert(!uniqueIDPtr);
+		macro_SetUniqueID(++maxUniqueID);
```

Comment:
This is making me question the check at line 169, honestly. And also, why are we bothering to set `uniqueIDPtr` to NULL?

Reply:
I originally removed the `if (uniqueID > maxUniqueID)` check (I believe it's not necessary in `master` either), but wanted to keep this diff minimal.

Reply:
I guess now that this isn't blindly returning `uniqueIDPtr`'s value, we don't need to set it to `NULL`.